smolyay/basis.py

```diff
@@ -328,141 +278,138 @@ def derivative(self, x):
-                    points.append(p)
-        return NestedBasisFunctionSet(points,basis_functions,levels)
-
-
-class BasisFunctionSet():
-    """Set of basis functions and sample points.
 
     Parameters
     ----------
-    basis_functions : list
-        Basis functions in set.
-
-    points : list
-        Sample point corresponding to each basis function.
+    n : int
+        Degree of trigonometric polynomial.
 
     """
 
-    def __init__(self,points,basis_functions):
-        self._basis_functions = basis_functions
-        self._points = points
-        if len(basis_functions) != len(points):
-            raise IndexError("basis_functions and points must have the "
-                    "same number of elements.")
+    def __init__(self, n):
+        super().__init__()
+        self.n = n
 
     @property
-    def points(self):
-        """list: Sampling points."""
-        return self._points
+    def domain(self):
+        """numpy.ndarray: Domain the sample points come from."""
+        return numpy.array([0, 2 * numpy.pi])
 
     @property
-    def basis_functions(self):
-        """list: Basis functions."""
-        return self._basis_functions
+    def sigma(self):
+        """float: Frequency of the complex exponential."""
+        if self.n % 2 == 1:
+            return (1 + self.n) / 2
+        else:
+            return -self.n / 2
 
+    def _function(self, x):
+        r"""Evaluate the basis function.
 
-class NestedBasisFunctionSet(BasisFunctionSet):
-    """Nested set of basis functions and points.
+        The Trigonometric polynomial is evaluated using the following formula:
 
-    Nested points/basis function grow in levels, such that an approximation
-    of a given level uses not only its sampling points but also all the points
-    at lower levels. Nested sets (similarly to ogres) are like onions.
+        .. math::
 
-    Parameters
-    ---------
-    levels : list of lists
-       Assignment of points/basis functions to each level.
-    """
+            \phi_n(x) = \exp(x i \sigma(n))
 
-    def __init__(self,points,basis_functions,levels):
-        super().__init__(points,basis_functions)
-        self._levels = levels
+        where *n* is the order of trigonometric polynomial and
+        is a non-negative integer, and `sigma` the frequency of the
+        complex exponential
+
+        Parameters
+        ----------
+        x : float
+            One-dimensional point on :math:`[0, 2\pi]`.
+
+        Returns
+        -------
+        float
+            Value of Trigonometric polynomial.
 
-    @property
-    def levels(self):
-        """list: List of lists of indexes for points/functions at each level.
         Raises
         ------
-        IndexError
-            max index must be less than total number of points.
+        ValueError
+            If input is outside the domain `[0, 2\pi]`
+        """
+        x = numpy.asarray(x)
+        return numpy.exp(x * self.sigma * 1j)
```

Comment:
I would expect this basis function to use `n` here (positive or negative) rather than `sigma`. I think `sigma` is more like an internal variable that would be used to nest these. Please confirm with Mohammadreza, and then probably remove `sigma` from this class

Reply:
He said it's okay to remove it.